plugins/funind/glob_term_to_relation.ml

```diff
@@ -630,6 +632,7 @@ let rec build_entry_lc env funnames avoid rt : glob_constr build_entry_return =
         let v = match typ with None -> v | Some t -> DAst.make ?loc:rt.loc @@ GCast (v,CastConv t) in
 	let v_res = build_entry_lc env funnames avoid v in
 	let v_as_constr,ctx = Pretyping.understand env (Evd.from_env env) v in
+        let v_as_constr = EConstr.Unsafe.to_constr v_as_constr in
 	let v_type = Typing.unsafe_type_of env (Evd.from_env env) (EConstr.of_constr v_as_constr) in
```

Comment:
Remove the `EConstr.Unsafe.to_constr` and the `EConstr.of_constr` (and use `EConstr push_named` line 638).